### DISCUSSION TITLE
Structure of Demo Code

### DISCUSSION THREAD

**Discussion by AndyTGardner** [Megabit/Blazorise#4382] [General]
I am new to Blazor -- so in all likelihood, this question reflects my ignorance. As I try to convert my windows form app to use Blazor/Blazorise, I am unclear to what extent the demo code structure reflects an elegant solution to testing vs an unnecessary complication.  I don't mean to criticize; I'm just trying to understand how to structure my app. Specifically, as I trace the code Components/SideMenu.razor => /tests/charts which aliases to Pages/Tests/ChartsPage.razor , etc.

Again, forgive my ignorance, but could someone give me some insight into whether this structure is idiosyncratic to this demo or recommended practice?  Also, I fail to see how the To = "/tests/charts" directs to ChartsPage.razor -- any insight would be appreciated.

Thanks in advance.

**Comment by stsrki** [comment]
Hi @AndyTGardner, our demo doesn't follow any guidelines on structuring real-world apps. You can take some parts as inspiration. But overall, demos are mainly used to test and see if our components are working correctly quickly. For example, form input can be in horizontal vs. vertical mode, grid layouts, etc., and those are easy to break. So we tend to focus most of our tests there. Since demo apps follow the recommended structure of the basic Blazorise components, that is the best way to test for us. In conclusion, you should use our demos as a reference for structuring component blocks. But for overall application design, there are better sources put there.

`To = "/tests/charts"` directs to the page that has a `@page "/tests/charts"` attribute defined.
